localstack/utils/cloudformation/template_deployer.py

```diff
@@ -2008,25 +2010,45 @@ def merge_properties(self, resource_id, old_stack, new_stack):
             "Resources"
         ][resource_id]
 
-    def apply_parameter_changes(self, old_stack, new_stack):
+    def resolve_param(
+        self, logical_id: str, param_type: str, default_value: Optional[str] = None
+    ) -> Optional[str]:
+        if param_type == "AWS::SSM::Parameter::Value<String>":
+            ssm_client = aws_stack.connect_to_service("ssm")
+            param = ssm_client.get_parameter(Name=default_value)
+            return param["Parameter"]["Value"]
+        return None
+
+    def apply_parameter_changes(self, old_stack, new_stack) -> None:
         parameters = {
-            p["ParameterKey"]: p["ParameterValue"] for p in old_stack.metadata["Parameters"]
+            p["ParameterKey"]: p
+            for p in old_stack.metadata["Parameters"]  # go through current parameter values
         }
 
-        for key, value in new_stack.template["Parameters"].items():
-            parameters[key] = value.get("Default", parameters.get(key))
+        for logical_id, value in new_stack.template["Parameters"].items():
+            default = value.get("Default")
```

Comment:
nit: tiny enhancement - call to `resolve_param(..)` is only required if `default` is `None`:
```
        default = value.get("Default")
        parameter_value = default
        if default is None:
            param_type = value.get("Type")
            provided_param_value = parameters.get(logical_id)
            resolved_value = self.resolve_param(logical_id, param_type, default)
            parameter_value = provided_param_value if default is None else default
```

Reply:
@whummer default actually usually needs to be defined for the resolving to work. The default in this case refers to the param name then for example